src/items.c

```diff
@@ -835,47 +839,79 @@ void weapon_touch()
 
 	if (!strcmp(self->classname, "weapon_nailgun"))
 	{
-		if (leave && ((int)other->s.v.items & IT_NAILGUN))
+		if ((leave || k_nosweep) && ((int)other->s.v.items & IT_NAILGUN))
```

Comment:
instead of checking k_nosweep set leave to 1 above if k_nosweep != 0

Reply:
seems I was wrong and it is a bit more complicated and leave flag used in more places than I expected